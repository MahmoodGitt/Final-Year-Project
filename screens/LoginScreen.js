// import {
// 	StyleSheet,
// 	TextInput,
// 	View,
// 	Text,
// 	TouchableOpacity,
// 	Alert,
// } from 'react-native';
// import React, { useState } from 'react';
// import auth from '../firebase/config';
// import { createUserWithEmailAndPassword } from 'firebase/auth';
// // import CustomstartScreenButtons from '../utilis/CustomstartScreenButtons';

// const LoginScreen = ({ navigation }) => {
//

import React, { useState } from 'react';
import {
	View,
	Text,
	TouchableOpacity,
	TextInput,
	Platform,
	StyleSheet,
	StatusBar,
	Alert,
	Image,
} from 'react-native';
import * as Animatable from 'react-native-animatable';
import { LinearGradient } from 'expo-linear-gradient';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import Feather from 'react-native-vector-icons/Feather';
import { Card } from 'react-native-paper';

/**
 * This functional compoenent contains the logic that handels the log-in process
 * @param {navigation}
 * @returns
 */
const LoginScreen = ({ navigation }) => {
	const [data, setData] = useState({
		username: '',
		password: '',
		check_textInputChange: false,
		secureTextEntry: true,
		isValidUserName: true,
		isValidPassword: true,
	});

	const textInputChange = (text) => {
		if (text.trim().includes('edu')) {
			setData({
				...data,
				username: text,
				check_textInputChange: true,
				isValidUserName: true,
			});
		} else {
			setData({
				...data,
				username: text,
				check_textInputChange: false,
				isValidUserName: false,
			});
		}
	};

	const handlePasswordChange = (val) => {
		if (val.trim().length >= 8) {
			setData({
				...data,
				password: val,
				isValidPassword: true,
			});
		} else {
			setData({
				...data,
				password: val,
				isValidPassword: false,
			});
		}
	};

	const updateSecureTextEntry = () => {
		setData({
			...data,
			secureTextEntry: !data.secureTextEntry,
		});
	};

	const handleValidUser = (text) => {
		if (text.includes('@edu')) {
			setData({
				...data,
				isValidUserName: true,
			});
			// console.log('pass');
		} else {
			setData({
				...data,
				isValidUserName: false,
			});
			// console.log('fail');
		}
	};

	return (
		<View style={styles.container}>
			<Card>
				<Card.Cover
					style={{ height: 200 }}
					source={{ uri: 'https://picsum.photos/700' }}
				/>
			</Card>
			<View style={styles.headerstyle}>
				<Text style={styles.text_header}>Welcome!</Text>
			</View>
			<Animatable.View animation="fadeInUpBig" style={styles.footerStyle}>
				<Text style={styles.text_footer}>Username</Text>
				<View style={styles.action}>
					<FontAwesome name="user-o" size={20} />
					<TextInput
						placeholder="Your Username"
						placeholderTextColor="#666666"
						style={styles.textInput}
						autoCapitalize="none"
						onChangeText={(text) => textInputChange(text)}
						onEndEditing={(endText) =>
							handleValidUser(endText.nativeEvent.text)
						}
					/>
					{data.check_textInputChange ? (
						<Animatable.View animation="bounceIn">
							<Feather name="check-circle" color="green" size={20} />
						</Animatable.View>
					) : null}
				</View>
				{data.isValidUserName ? null : (
					<Animatable.View animation="fadeInLeft" duration={500}>
						<Text style={styles.errorMsg}>
							Username must be your organisation's email.
						</Text>
					</Animatable.View>
				)}

				<Text
					style={[
						styles.text_footer,
						{
							marginTop: 35,
						},
					]}
				>
					Password
				</Text>
				<View style={styles.action}>
					<Feather name="lock" size={20} />
					<TextInput
						placeholder="Your Password"
						placeholderTextColor="#666666"
						secureTextEntry={data.secureTextEntry ? true : false}
						style={styles.textInput}
						autoCapitalize="none"
						onChangeText={(val) => handlePasswordChange(val)}
					/>
					<TouchableOpacity onPress={updateSecureTextEntry}>
						{data.secureTextEntry ? (
							<Feather name="eye-off" color="grey" size={20} />
						) : (
							<Feather name="eye" color="grey" size={20} />
						)}
					</TouchableOpacity>
				</View>
				{data.isValidPassword ? null : (
					<Animatable.View animation="fadeInLeft" duration={500}>
						<Text style={styles.errorMsg}>
							Password must be 8 characters long.
						</Text>
					</Animatable.View>
				)}

				<TouchableOpacity>
					<Text style={{ color: '#009387', marginTop: 15 }}>
						Forgot password?
					</Text>
				</TouchableOpacity>
				<View style={styles.startScreenButton}>
					<TouchableOpacity
						style={styles.logIn}
						// onPress={() => {loginHandle( data.username, data.password )}}
						onPress={() => {
							navigation.navigate('                    ');
						}}
					>
						<LinearGradient
							colors={['#08d4c4', '#01ab9d']}
							style={styles.logIn}
						>
							<Text style={styles.textSign}>Sign In</Text>
						</LinearGradient>
					</TouchableOpacity>

					<TouchableOpacity
						onPress={() => navigation.navigate('SignUp')}
						style={[
							styles.logIn,
							{
								borderColor: '#009387',
								borderWidth: 1,
								marginTop: 15,
							},
						]}
					>
						<Text
							style={[
								styles.textSign,
								{
									color: '#009387',
								},
							]}
						>
							Sign Up
						</Text>
					</TouchableOpacity>
				</View>
			</Animatable.View>
		</View>
	);
};

export default LoginScreen;

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: '#488529',
	},

	headerstyle: {
		flex: 0.3,
		justifyContent: 'center',
		alignItems: 'center',
	},

	headerText: {
		justifyContent: 'center',
		alignItems: 'center',
	},

	footerStyle: {
		flex: 3,
		backgroundColor: '#fff',
		borderTopLeftRadius: 30,
		borderTopRightRadius: 30,
		paddingHorizontal: 20,
		paddingVertical: 30,
	},
	text_header: {
		color: '#fff',
		fontWeight: 'bold',
		fontSize: 30,
		alignItems: 'center',
		justifyContent: 'center',
	},
	text_footer: {
		color: '#05375a',
		fontSize: 18,
	},
	action: {
		flexDirection: 'row',
		marginTop: 10,
		borderBottomWidth: 1,
		borderBottomColor: '#f2f2f2',
		paddingBottom: 5,
	},
	actionError: {
		flexDirection: 'row',
		marginTop: 10,
		borderBottomWidth: 1,
		borderBottomColor: '#FF0000',
		paddingBottom: 5,
	},

	logIn: {
		width: '100%',
		height: 50,
		justifyContent: 'center',
		alignItems: 'center',
		borderRadius: 10,
	},

	textInput: {
		flex: 1,
		marginTop: Platform.OS === 'ios' ? 0 : -12,
		paddingLeft: 10,
		color: '#05375a',
	},
	errorMsg: {
		fontSize: 14,
		color: '#FF0000',
	},

	textSign: {
		fontSize: 18,
		fontWeight: 'bold',
	},
	startScreenButton: {
		alignItems: 'center',
		marginTop: 50,
	},
});
