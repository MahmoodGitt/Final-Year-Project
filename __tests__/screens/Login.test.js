/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';

import LoginScreen from '../../screens/LoginScreen';

describe('Log in Component', () => {
	it('Testing if Login Screen Component loads up', () => {
		const tree = renderer.create(<LoginScreen />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
