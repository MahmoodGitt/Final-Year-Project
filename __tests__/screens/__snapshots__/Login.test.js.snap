// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up Component Testing if Sign up Screen Component loads up 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#488529",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "rgba(0, 0, 0, 0.12)",
        "borderRadius": 4,
        "elevation": 1,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.5,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 0.75,
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexGrow": 1,
          "flexShrink": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#eeeeee",
              "height": 195,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
            Object {
              "height": 200,
            },
          ]
        }
      >
        <Image
          siblings={
            Array [
              "withTheme(Card.Cover)",
            ]
          }
          source={
            Object {
              "uri": "https://picsum.photos/700",
            }
          }
          style={
            Array [
              Object {
                "flex": 1,
                "height": undefined,
                "justifyContent": "flex-end",
                "padding": 16,
                "width": undefined,
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    forwardedRef={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 3,
        "opacity": 0,
        "paddingHorizontal": 20,
        "paddingVertical": 30,
        "transform": Array [
          Object {
            "translateY": 500,
          },
        ],
      }
    }
  >
    <RCTScrollView
      collapsable={false}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "marginVertical": 50,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#05375a",
                "fontSize": 18,
              }
            }
          >
            Email
          </Text>
          <View
            style={
              Object {
                "borderBottomColor": "#f2f2f2",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "marginTop": 10,
                "paddingBottom": 5,
              }
            }
          >
            <Text />
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              onChangeText={[Function]}
              placeholder="Your Email"
              placeholderTextColor="#666666"
              rejectResponderTermination={true}
              style={
                Object {
                  "color": "#05375a",
                  "flex": 1,
                  "marginTop": 0,
                  "paddingLeft": 10,
                }
              }
              underlineColorAndroid="transparent"
            />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#05375a",
                  "fontSize": 18,
                },
                Object {
                  "marginTop": 35,
                },
              ]
            }
          >
            Password
          </Text>
          <View
            style={
              Object {
                "borderBottomColor": "#f2f2f2",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "marginTop": 10,
                "paddingBottom": 5,
              }
            }
          >
            <Text />
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              onChangeText={[Function]}
              placeholder="Your Password"
              placeholderTextColor="#666666"
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Object {
                  "color": "#05375a",
                  "flex": 1,
                  "marginTop": 0,
                  "paddingLeft": 10,
                }
              }
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 50,
              "marginHorizontal": 80,
              "width": "50%",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#009387",
                "borderRadius": 10,
                "borderWidth": 1,
                "height": 50,
                "justifyContent": "center",
                "opacity": 1,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#009387",
                  "fontSize": 18,
                  "fontWeight": "bold",
                }
              }
            >
              Log in
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginTop": 30,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#009387",
              }
            }
          >
            Do you not have an account?
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "blue",
                  "marginLeft": 10,
                }
              }
            >
              Click Here
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
